### ISSUE TITLE
flux-jobs: Support verbose default option

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#2627]
As time goes on, we will learn about more information that users like to get, but just not as much as the default output.  Some possibilities:

- exit code
- priority
- node count
- time limit

**Comment by chu11** [comment]
from #4594

just to get the ball rolling on ideas for `--format=long`, I'll throw this out as a contender

```
flux jobs -a -o '{id.f58:>12} {username:<8.8} {name:<10.10} {state:>8} {result:>9h} {ntasks:>6} {nnodes:>6h} {runtime\!F:>8h} {t_remaining\!F:>12h} {nodelist:h}'
```

basically status_abbrev -> state & result (no abbrev, full names)
add t_remaining

initially added expiration, but felt it had less value.

figured priority wouldn't matter to average user?

TBH, this isn't much more than the default. Although I think `t_remaining` is useful for running jobs.

**Comment by grondo** [comment]
I'd suggest the addition of `t_submit`, perhaps `{t_submit!d:%b%d %R}` (except that padding and alignment are supported for `datetime` formats. Hmm, maybe introduction of a new conversion specifier that converts a datetime to this form would be useful)

Also, I don't find that separating `{status}` into `{state}` and `{result}` adds much. Maybe just replace `{status_abbrev}` with the full `{status}`?

Finally, I would also suggest adding `sched.t_estimate` when it exists. Maybe we could introduce a new format field that returns `t_estimate` for pending jobs or the nodelist for other jobs.

**Comment by chu11** [comment]
> Maybe just replace {status_abbrev} with the full {status}?

Yeah I agree.

> I'd suggest the addition of t_submit

This surprises me, what is the average use case / interest?  Perhaps to try and cancel jobs submitted a long time ago?

> Finally, I would also suggest adding sched.t_estimate when it exists.

Oh yeah, that's a good idea.  We don't list it in`flux-jobs(1)` so totally forgot about it.  Especially once the #4591 goes in so it works with schedulers that don't support it.

**Comment by grondo** [comment]
> This surprises me, what is the average use case / interest? Perhaps to try and cancel jobs submitted a long time ago?

Well otherwise you have no idea how long a job has been sitting in the queue or "been around". You might see 5 jobs in the queue, but not know that one has been there for 5 days. This would be the equivalent to the `ps(1)` `START` column which is displayed with `ps aux`. Given that our model for `flux jobs` was `ps` in some ways, it seems reasonable to include this field.

**Comment by chu11** [comment]
Makes sense, how about (didn't have your #4597 pulled in yet, would add that later)

`flux jobs -a -o '{id.f58:>12} {username:<8.8} {name:<10.10} {status:>9} {ntasks:>6} {nnodes:>6h} {runtime\!F:>8h} {t_submit\!d:%b%d %R} {t_remaining\!F:>12h} {nodelist:h}'`

```
      JOBID USER     NAME          STATUS NTASKS NNODES  RUNTIME T_SUBMIT  T_REMAINING NODELIST
 ƒCMxV3cnTBV achu     sleep      COMPLETED      1      1   1.688m Sep21 12:38            - corona171
 ƒCMw4Wi7PGb achu     sleep      COMPLETED      1      1   1.688m Sep21 12:35            - corona171
 ƒCMvAHY8p87 achu     sleep      COMPLETED      1      1   1.021m Sep21 12:33            - corona171
```

We could add `sched.t_estimate` now??  With the "collapse if empty" support soon going in, it would be hidden until its supported.

**Comment by grondo** [comment]
Yeah, I was going to propose something similar (also including `?:{queue}` of course). Seems like a very good start!

**Comment by chu11** [comment]
> Yeah, I was going to propose something similar (also including ?:{queue} of course). Seems like a very good start!

A little off topic to this thread (we can start a new issue), but shouldn't ?:{queue} be in the default output as well?  Seems important enough.

**Comment by grondo** [comment]
I had that on my list, but there wasn't an issue (I think it was lumped in with "flux-jobs: support job queue"). I opened an issue. #4602

**Comment by grondo** [comment]
> hmmm was I omniscient?

The english teacher in me wants you to say prescient :laughing:
